### ISSUE TITLE
TF 2.0  ft.GradientTape() gradient()  second gradient None

### ISSUE THREAD

**Issue by zwenju** [tensorflow/tensorflow#31519]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows Anaconda 
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below): tensorflow 2.0b1
- Python version:  3.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
second gradient with input  is None

**Describe the expected behavior**

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
```
import tensorflow as tf
import numpy as np
from matplotlib import pyplot as plt
import time
import math
import tensorflow as tf
import numpy as np
import tensorflow as keras
from tensorflow.keras import layers
from tensorflow import keras
from tensorflow.keras.utils import plot_model
from scipy . stats import multivariate_normal as normal

d = 2
batch_size = 3


def func(x):
    x = x*x
    x = keras.layers.Dense(d)(x)
    return x

Input  = keras.Input(shape = (d,), dtype = tf.float64, name = 'X' )
Output = func(Input)
model = keras.Model(inputs=Input, outputs=Output)



x_train = tf.random.uniform( (batch_size, d), minval=0, maxval=1, dtype=tf.float64, seed=1000)

for epoch in range (1):
    with tf.GradientTape() as t1:
        t1.watch(x_train)
        with tf.GradientTape() as t2:
            t2.watch(x_train)
            predictions = model( x_train )

        dy_dx = t2.gradient(predictions, x_train)
        print("dy_dx = ", dy_dx)
    dyy_dx = t1.gradient(dy_dx, x_train)

    print("dyy_dx = ", dyy_dx)
```
**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
W0811 23:13:04.473541 11804 ag_logging.py:145] Entity <bound method TensorFlowOpLayer._defun_call of <tensorflow.python.eager.function.TfMethodTarget object at 0x0000025C98A045F8>> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method TensorFlowOpLayer._defun_call of <tensorflow.python.eager.function.TfMethodTarget object at 0x0000025C98A045F8>>: AssertionError: 
WARNING: Entity <bound method TensorFlowOpLayer._defun_call of <tensorflow.python.eager.function.TfMethodTarget object at 0x0000025C98A045F8>> could not be transformed and will be executed as-is. Please report this to the AutoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method TensorFlowOpLayer._defun_call of <tensorflow.python.eager.function.TfMethodTarget object at 0x0000025C98A045F8>>: AssertionError: 
dy_dx =  tf.Tensor(
[[0.03395048 0.86635576]
 [0.52642456 0.80100264]
 [1.05554004 0.05975098]], shape=(3, 2), dtype=float64)
dyy_dx =  None

**Comment by oanush** [comment]
@zwenju ,
Hi,When tried executing the given code error `NameError: name 'neutron_network' is not defined` was faced.Thanks!

**Comment by zwenju** [comment]
Please , try the new code,  I have change neutron_network to func.  thanks

**Comment by oanush** [comment]
@zwenju,
Thank you for the code,when tried executing the given code i got the output as per the screenshot
![output](https://user-images.githubusercontent.com/52397990/63010108-667e2300-bea3-11e9-958e-b8a9a8e843f5.png)

**Comment by zwenju** [comment]
Thank you,      **dyy_dx = None**  is  actually not the expect results.
Could you fix this bug ?
Thanks.

**Comment by ymodak** [comment]
This is fixed with latest version TF 2.0 nightly
Output in TF 2.X nightly version 2.0.0-dev20190821
```python
dy_dx =  tf.Tensor(
[[ 2.84032801 -0.34441594]
 [ 0.12755782 -0.69508436]
 [ 1.86527439 -0.02203789]], shape=(3, 2), dtype=float64)
dyy_dx =  tf.Tensor(
[[ 3.79678631 -0.80638027]
 [ 3.79678631 -0.80638027]
 [ 3.79678631 -0.80638027]], shape=(3, 2), dtype=float64)
```

**Comment by zwenju** [comment]
Great Thanks.
